src/WebJobs.Script.WebHost/Management/VirtualFileSystem.cs

```diff
@@ -641,10 +640,40 @@ private IEnumerable<VfsStatEntry> GetDirectoryResponse(HttpRequest request, File
         protected HttpResponseMessage CreateResponse(HttpStatusCode statusCode, object payload = null)
         {
             var response = new HttpResponseMessage(statusCode);
-            if (payload != null)
+            try
+            {
+                if (payload != null)
+                {
+                    // Use safe serialization settings
+                    var jsonSerializerSettings = new JsonSerializerSettings
+                    {
+                        NullValueHandling = NullValueHandling.Ignore,
+                        DefaultValueHandling = DefaultValueHandling.Include,
+                        Formatting = Formatting.None
+                    };
+
+                    // Sanitize the payload if it's an object
```

Comment:
I did not get where are we sanitizing here. Am I missing something?

Reply:
Checking the payload object is exception to remove the details